Xamarin.PropertyEditing/ViewModels/RatioViewModel.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Xamarin.PropertyEditing.Drawing;
+
+namespace Xamarin.PropertyEditing.ViewModels
+{
+	internal class RatioViewModel
+		: PropertyViewModel<CommonRatio>
+	{
+		public double Numerator
+		{
+			get { return Value.Numerator; }
+			set {
+				if (Value.Numerator.Equals (value))
+					return;
+
+				Value = new CommonRatio (value, Value.Denominator, Value.RatioSeparator);
+			}
+		}
+
+		public double Denominator
+		{
+			get { return Value.Denominator; }
+
+			set {
+				if (Value.Denominator.Equals (value))
+					return;
+
+				Value = new CommonRatio (Value.Numerator, value, Value.RatioSeparator);
+			}
+		}
+
+		public char RatioSeparator {
+			get { return Value.RatioSeparator; }
+
+			set {
+				if (Value.RatioSeparator.Equals (value))
+					return;
+
+				Value = new CommonRatio (Value.Numerator, Value.Denominator, value);
+			}
+		}
+
+		internal void ValueChanged (string stringValue, int caretPosition, int selectionLength, double incrementValue)
+		{
+			var separator = GetSeparator (stringValue);
+			SetRatioFromString (stringValue);
+
+			var newNumerator = Value.Numerator + incrementValue;
+			if (newNumerator < 1)
+				newNumerator = 1;
+
+			var newDenominator = Value.Denominator + incrementValue;
+			if (newDenominator < 1)
+				newDenominator = 1;
+
+			// Full selection, increment both
+			if (selectionLength == stringValue.Length) {
+				// Increment both values.
+				Value = new CommonRatio (newNumerator, newDenominator, separator);
+			} else {
+				// Only Increment numerator 
+				Value = new CommonRatio (newNumerator, Value.Denominator, separator);
+			}
+		}
+
+		public string ValueString
+		{
+			get { return Value.StringValue; }
+			set {
+				if (Value.StringValue.Equals (value))
+					return;
+
+				SetRatioFromString (value);
+			}
+		}
+
+		public static char[] Separators = { ':', '/' };
+		public static char[] SplitSeparators = { ':', '/', ' ' };
```

Comment:
I don't think that we want to accept spaces.

Reply:
This was to more easily split should the user enter something like " 21: 2 ". In theory the `double.TryParse()` check should handle it. But if you feel strongly about it, I can remove it.